### PR TITLE
solution

### PR DESCRIPTION
**Pull request by savchuksavchuk**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,42 @@
...
+          delete temp[remove[index]];
+        }
+        arr.push({ ...temp });
+        break;
+
+      case action.type === 'clear':
+        for (const key in temp) {
+          delete temp[key];
+        }
+        arr.push({ ...temp });
```
Comment (valerii-derkach) [inline src/transformStateWithClones.js:40]: -If you perform same action in all switch cases - do it just once afterwards.
-name variables meanful
checklist 4,5,6 https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md

**Review by valerii-derkach — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,41 @@
...
+  Object.assign(copyState, state);
+
+  const result = [];
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
+        const { extraData } = action;
+
+        for (const index in extraData) {
```
Comment (valerii-derkach) [inline src/transformStateWithClones.js:21]:
```suggestion
    switch (action.type) {
      case 'addProperties':
        const { extraData } = action;

        for (const index in extraData) {
```

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,41 @@
...
+
+      case action.type === 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const index in keysToRemove) {
+          delete copyState[keysToRemove[index]];
+        }
+        break;
+
+      case action.type === 'clear':
```
Comment (valerii-derkach) [inline src/transformStateWithClones.js:34]:
```suggestion
      case  'clear':
```

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,41 @@
...
+        for (const index in extraData) {
+          copyState[index] = extraData[index];
+        }
+        break;
+
+      case action.type === 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const index in keysToRemove) {
+          delete copyState[keysToRemove[index]];
```
Comment (valerii-derkach) [inline src/transformStateWithClones.js:30]:
```suggestion

      'removeProperties':
        const { keysToRemove } = action;

        for (const index in keysToRemove) {
          delete copyState[keysToRemove[index]];
```

**Review by valerii-derkach — CHANGES_REQUESTED** [review]

**Review by marina-tilniak — CHANGES_REQUESTED** [review]
add default branch

**Review by marina-tilniak — APPROVED** [review]

**Pull request closed without merging**